### PR TITLE
treewide: Initial role commit

### PR DESCRIPTION
**Pull request by haslersn**
_I had to recreate #1_

- [x] `README.md`
- [x] Tested to be working

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
+# librespot
+
+This role installs [librespot](https://github.com/librespot-org/librespot) and creates a
```
Comment (SlothOfAnarchy) [inline README.md:3]: one full sentence per line

File: README.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+## Requirements
+
+An apt based system like [Ubuntu](https://www.ubuntu.com/) or [Debian](https://www.debian.org/).
+
+## Updating
+
+This role can also update librespot to a desired version.
+To do so, just change the [source specification](#source-specification) and the `librespot_version`
+variable accordingly, and then reapply this role.
```
Comment (SlothOfAnarchy) [inline README.md:15]: .

File: README.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+| librespot_zeroconf_port               | _undefined_          | If defined, argument passed to `--zeroconf-port`            |
+| librespot_normalisation_pregain       | _undefined_          | If defined, argument passed to `--normalisation-pregain`    |
+| librespot_extra_options               | []                   | List of strings (flags or options) passed to librespot      |
+
+### Source Specification
+
+There are multiple ways to specify the source from which librespot should be downloaded.
+Chosing among them is done by providing exactly one of the following role variables.
+Actually, multiple of those role variables can be provided, but then only the one listed here first
+is considered.
```
Comment (SlothOfAnarchy) [inline README.md:62]: .

File: README.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+* `librespot_binary_url`:
+  Download a precompiled binary from that URL.
+  **You should anyway set `librespot_version` to a value that's unique for that version.**
+  This is because librespot is only re-downloaded, when the `librespot_version` role variable
+  changes.
```
Comment (SlothOfAnarchy) [inline README.md:68]: .

File: README.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+* `librespot_archive_url`:
+  Build from source.
+  The source is downloaded as an archive from that URL.
+  The downloaded archive must contain a folder named `librespot-{{ librespot_version }}` which
+  contains in turn the source that can be compiled with `cargo build`.
```
Comment (SlothOfAnarchy) [inline README.md:73]: .

File: meta/main.yml [hunk 6]
```diff
@@ -0,0 +1,21 @@
+---
+galaxy_info:
+  author: haslersn
+  description: Role for https://github.com/librespot-org/librespot
```
Comment (SlothOfAnarchy) [inline meta/main.yml]: Please the scribe what librespot is, instead of the URL

File: meta/main.yml [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+  company: stuvus (https://stuvus.uni-stuttgart.de)
+
+  license: MIT
+
+  min_ansible_version: 2.4
+
+  platforms:
+    - name: Debian
```
Comment (SlothOfAnarchy) [inline meta/main.yml:12]: Which versions? Not all support e.g. systemd

**Review by SlothOfAnarchy — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (haslersn) [reply README.md:3]: That would exceed the column limit in this case.

File: meta/main.yml (hunk 6 above)
Comment (haslersn) [reply meta/main.yml]: It is described by the librespot project owners at that URL. The project owners are the best at describing what the project does.

File: meta/main.yml (hunk 7 above)
Comment (haslersn) [reply meta/main.yml:12]: I don't have time to test it on all Debian versions, so I added the one I tested it on.

**Comment by Scriptkiddi** [comment]
@SlothOfAnarchy

**Review by SlothOfAnarchy — APPROVED** [review]

**Pull request merged**